## Data Processing: Coral Reefs 

Group 3 
Dani Madan, Dina Millerman, Jesse Levisohn

Read in data: 

In [119]:
import numpy as np 
import pandas as pd 

In [120]:
Cardbonate_Chemistry = '/Users/danimadan/Desktop/GitHub/group3-project/FINAL Cleaned Carbonate Chemistry, water quality and coral measurements.xlsx'
data= pd.ExcelFile(Cardbonate_Chemistry)

#generating list of sheets 
sheet_names = data.sheet_names

print(sheet_names)


['pH control', 'pH Elevated PCO2', 'Water Quality', 'Mcav Control', 'Mcav Elevated PCO2', 'Ofav Control', 'Ofav Elevated P CO2', 'Pcli Elevated PCO2', 'Pcli Control']


In [121]:
#loading all sheets as seperate dataframe objects
ph_control = pd.read_excel(data, sheet_name = 'pH control')

ph_elevated_pc02 = pd.read_excel(data, sheet_name ='pH Elevated PCO2')

water_quality = pd.read_excel(data, sheet_name ='Water Quality')

mcav_control = pd.read_excel(data, sheet_name ='Mcav Control')

mcav_elevated_PC02 = pd.read_excel(data, sheet_name ='Mcav Elevated PCO2')

ofav_control = pd.read_excel(data, sheet_name ='Ofav Control')

ofav_evelavted_p_c02 = pd.read_excel(data, sheet_name = 'Ofav Elevated P CO2')

pcli_control_pc02 = pd.read_excel(data, sheet_name = 'Pcli Control')

pcli_evelvated_pc02 = pd.read_excel(data, sheet_name = 'Pcli Elevated PCO2')



Did each sheet seperately so data frames could have different names 

**pH Control**

In [122]:
# basic summary, shape, 
print('rows', len(ph_control))
print()

print('shape:', ph_control.shape)
print()

print(ph_control.columns)
print()

print('data types: ', ph_control.dtypes)



rows 112

shape: (112, 4)

Index(['Week', 'pH', 'Total Alkalinity (µmol/kg SW)  ', 'pCO2 (µatm)'], dtype='object')

data types:  Week                                 int64
pH                                 float64
Total Alkalinity (µmol/kg SW)      float64
pCO2 (µatm)                        float64
dtype: object


112 obersvations total, helps get a sense of how much data is missing and if this will impact findings

In [123]:
#summary statistics pH control 
ph_control.describe()


Week          pH  Total Alkalinity (µmol/kg SW)    pCO2 (µatm)
count  112.000000  110.000000                       103.000000   103.000000
mean     6.035714    8.017779                      2953.776505   560.086767
std      3.646369    0.051584                       369.062846    97.248136
min      1.000000    7.729335                      2440.754000   455.173000
25%      3.000000    7.981825                      2548.868500   522.106500
50%      6.000000    8.018388                      2981.454000   546.584000
75%      9.000000    8.045537                      3171.456500   580.718500
max     12.000000    8.123369                      3822.653000  1422.321000

In [124]:
#missing data pH control 
print(ph_control.isnull().sum()) 

Week                               0
pH                                 2
Total Alkalinity (µmol/kg SW)      9
pCO2 (µatm)                        9
dtype: int64


**pH Elevated PC02**

In [125]:
#summmary statistics pH elevated 
ph_elevated_pc02.describe()

Week          pH  Total Alkalinity (µmol/kg SW)    pCO2 (µatm)
count  112.000000  106.000000                       102.000000    96.000000
mean     6.053571    7.802512                      2947.675382  1040.707813
std      3.673227    0.117780                       366.099921   211.833872
min      1.000000    7.569924                      2472.392000   635.181000
25%      3.000000    7.718991                      2550.512750   882.392500
50%      6.000000    7.818064                      2970.211500  1048.462000
75%      9.000000    7.869483                      3164.477250  1152.385250
max     12.000000    8.102513                      3824.921000  1617.739000

In [126]:
#missing date pH elevated 
print(ph_elevated_pc02.isnull().sum()) 

Week                                0
pH                                  6
Total Alkalinity (µmol/kg SW)      10
pCO2 (µatm)                        16
dtype: int64


**Water Quality**

In [127]:
#summmary statistics water quality 
water_quality.describe()

Temp (°C)  Salinity (am)  Salinity (pm)  Unnamed: 4  Unnamed: 5  \
count  95.000000      95.000000      65.000000         0.0         0.0   
mean   25.491465      34.643739      34.981538         NaN         NaN   
std     2.629227       3.586744       0.058342         NaN         NaN   
min     0.228958       0.055862      34.800000         NaN         NaN   
25%    25.600000      35.000000      35.000000         NaN         NaN   
50%    25.700000      35.000000      35.000000         NaN         NaN   
75%    25.900000      35.000000      35.000000         NaN         NaN   
max    26.200000      35.100000      35.100000         NaN         NaN   

       Ca 2+ (ppm)     Mg (ppm)  PO4 (ppm)  NH3 (ppm)  NO3 (ppm)  
count    15.000000    15.000000   9.000000   9.000000        9.0  
mean    430.905366  1246.728461   0.019592   0.003082        0.0  
std     115.408767   335.334077   0.014195   0.004279        0.0  
min      18.195872    43.234602   0.000000   0.000000        0.0  
25%     447.500000  1297.500000   0.010000   0.000000        0.0  
50%     455.000000  1320.000000   0.020000   0.000000        0.0  
75%     475.000000  1357.500000   0.020000   0.004880        0.0  
max     490.000000  1410.000000   0.050000   0.010000        0.0

In [128]:
#missing data water quality 
print(water_quality.isnull().sum()) 

Date              1
Temp (°C)         1
Salinity (am)     1
Salinity (pm)    31
Unnamed: 4       96
Unnamed: 5       96
Date.1           80
Ca 2+ (ppm)      81
Mg (ppm)         81
PO4 (ppm)        87
NH3 (ppm)        87
NO3 (ppm)        87
dtype: int64


**MCAV Control**

In [129]:
#summary statistics mcav control 
mcav_control.describe()

Day Measured  Buoyant Weight (g)  Tissue Surface Area (mm2)  \
count     40.000000           40.000000                  40.000000   
mean      42.000000            9.177375                 643.734137   
std       31.703757            2.063353                 185.207967   
min        0.000000            5.345000                 322.336520   
25%       21.000000            7.616000                 519.411795   
50%       42.000000            9.667000                 647.738202   
75%       63.000000           11.091750                 759.912072   
max       84.000000           11.696000                1012.217716   

       Surface Area Density (g/mm2)  Calcification rate (mgCaCO3/cm2/day)  \
count                     40.000000                             30.000000   
mean                      75.117365                              2.846270   
std                       31.656430                              2.841692   
min                       31.448997                             -3.400000   
25%                       51.017469                              1.582143   
50%                       72.222453                              2.612500   
75%                       87.644219                              3.875000   
max                      138.540905                              9.982143   

       Tissue growth (mm2/day)  Percent change Surface Area Density  Elevated  \
count                30.000000                            30.000000      40.0   
mean                  4.193819                            23.264035       0.0   
std                   3.715584                            27.124144       0.0   
min                  -1.425896                            -6.773841       0.0   
25%                   1.035460                             2.780162       0.0   
50%                   3.329976                            10.363548       0.0   
75%                   7.516959                            39.004207       0.0   
max                  10.371129                            87.680959       0.0   

          Sample  
count  40.000000  
mean    4.500000  
std     2.908872  
min     0.000000  
25%     2.000000  
50%     4.500000  
75%     7.000000  
max     9.000000

In [130]:
#missing data mcav control 
print(mcav_control.isnull().sum()) 

Day Measured                             0
Buoyant Weight (g)                       0
Tissue Surface Area (mm2)                0
Surface Area Density (g/mm2)             0
Calcification rate (mgCaCO3/cm2/day)    10
Tissue growth (mm2/day)                 10
Percent change Surface Area Density     10
Elevated                                 0
Sample                                   0
dtype: int64


**MCAV Elevated PC02**

In [131]:
#summary statistics mcav elevated 
mcav_elevated_PC02.describe()

Day Measured   Buoyant Weight (g)  Tissue Surface Area (mm2)  \
count      40.000000           40.000000                  40.000000   
mean       42.000000            8.742175                 591.842211   
std        31.703757            2.054346                 144.409360   
min         0.000000            5.072000                 324.182734   
25%        21.000000            7.206750                 521.980577   
50%        42.000000            9.405000                 622.675716   
75%        63.000000           10.111750                 687.689879   
max        84.000000           12.176000                 864.170603   

       Surface Area Density (g/mm2)  Calcification rate (mgCaCO3/cm2/day)  \
count                     40.000000                             30.000000   
mean                      71.050551                              2.939762   
std                       23.167985                              3.702228   
min                       33.956503                             -4.200000   
25%                       57.436811                              0.312500   
50%                       65.907090                              2.250000   
75%                       83.938919                              4.583929   
max                      127.150102                             13.732143   

       Tissue growth (mm2/day)  Percent change Surface Area Density  Elevated  \
count                30.000000                            30.000000      40.0   
mean                  3.657831                            21.673681       1.0   
std                   3.217451                            25.550538       0.0   
min                  -1.686665                            -5.071345       1.0   
25%                   1.097233                             2.791296       1.0   
50%                   2.886003                            11.068506       1.0   
75%                   6.113543                            41.904042       1.0   
max                  10.775937                            90.873656       1.0   

          Sample  
count  40.000000  
mean   14.500000  
std     2.908872  
min    10.000000  
25%    12.000000  
50%    14.500000  
75%    17.000000  
max    19.000000

In [132]:
#missing data mcav elevated 
print(mcav_elevated_PC02.isnull().sum()) 

Day Measured                             0
Buoyant Weight (g)                       0
Tissue Surface Area (mm2)                0
Surface Area Density (g/mm2)             0
Calcification rate (mgCaCO3/cm2/day)    10
Tissue growth (mm2/day)                 10
Percent change Surface Area Density     10
Elevated                                 0
Sample                                   0
dtype: int64


**OFAV Control**

In [133]:
#summary statistics OFAV Control 
ofav_control.describe()

Date data was taken  Buoyant Weight (g)  Tissue Surface Area (mm2)  \
count            40.000000           40.000000                  40.000000   
mean             42.000000            8.369100                 340.795261   
std              31.703757            1.327868                 113.202844   
min               0.000000            5.954000                 172.721117   
25%              21.000000            7.923250                 262.342831   
50%              42.000000            8.674500                 313.978559   
75%              63.000000            8.871750                 403.260403   
max              84.000000           10.841000                 604.948457   

       Surface Area Density (g/mm2)  Calcification rate (mgCaCO3/cm2/day)  \
count                     40.000000                             30.000000   
mean                      41.501356                              1.601706   
std                       14.170962                              5.597612   
min                       18.896226                             -6.866667   
25%                       30.476652                             -2.044643   
50%                       40.855871                              1.377976   
75%                       49.176904                              3.350893   
max                       76.161295                             22.083333   

       Tissue growth (mm2/day)  Percent change Surface Area Density  Elevated  \
count                30.000000                            30.000000      40.0   
mean                 -0.363665                            -1.921978       0.0   
std                   3.601780                            27.009731       0.0   
min                  -6.869236                           -44.488630       0.0   
25%                  -3.001731                           -20.154169       0.0   
50%                  -0.598413                            -5.016618       0.0   
75%                   1.410959                            16.931742       0.0   
max                   7.864161                            55.621099       0.0   

          Sample  
count  40.000000  
mean    4.500000  
std     2.908872  
min     0.000000  
25%     2.000000  
50%     4.500000  
75%     7.000000  
max     9.000000

In [134]:
#missing data OFAV control 
print(ofav_control.isnull().sum()) 

Date data was taken                      0
Buoyant Weight (g)                       0
Tissue Surface Area (mm2)                0
Surface Area Density (g/mm2)             0
Calcification rate (mgCaCO3/cm2/day)    10
Tissue growth (mm2/day)                 10
Percent change Surface Area Density     10
Elevated                                 0
Sample                                   0
dtype: int64


**Ofav Elevated PC02**

In [135]:
#summary statistics ofav elevated 
ofav_evelavted_p_c02.describe()

Date data was taken  Buoyant Weight (g)  Tissue Surface Area (mm2)  \
count            40.000000           40.000000                  40.000000   
mean             42.000000            8.531000                 379.814788   
std              31.703757            1.480309                 167.989748   
min               0.000000            6.198000                 185.671821   
25%              21.000000            7.436250                 244.288570   
50%              42.000000            8.304000                 294.121280   
75%              63.000000            9.315500                 523.605297   
max              84.000000           11.084000                 692.114617   

       Surface Area Density (g/mm2)  Calcification rate (mgCaCO3/cm2/day)  \
count                     40.000000                             30.000000   
mean                      43.993775                              1.075873   
std                       16.693181                              2.726150   
min                       21.454859                             -2.928571   
25%                       30.305107                             -0.531250   
50%                       40.038360                              0.585714   
75%                       56.493403                              2.104167   
max                       84.229599                              9.625000   

       Tissue growth (mm2/day)  Percent change Surface Area Density  Elevated  \
count                30.000000                            30.000000      40.0   
mean                  0.459133                             5.223983       1.0   
std                   4.491028                            35.723231       0.0   
min                 -10.103761                           -55.499926       1.0   
25%                  -1.618021                           -14.306509       1.0   
50%                   0.452066                            -0.355881       1.0   
75%                   2.174681                            16.618425       1.0   
max                   9.541872                           104.763664       1.0   

          Sample  
count  40.000000  
mean   14.500000  
std     2.908872  
min    10.000000  
25%    12.000000  
50%    14.500000  
75%    17.000000  
max    19.000000

In [136]:
#missing data ofav elevated 
print(ofav_evelavted_p_c02.isnull().sum()) 

Date data was taken                      0
Buoyant Weight (g)                       0
Tissue Surface Area (mm2)                0
Surface Area Density (g/mm2)             0
Calcification rate (mgCaCO3/cm2/day)    10
Tissue growth (mm2/day)                 10
Percent change Surface Area Density     10
Elevated                                 0
Sample                                   0
dtype: int64


**PCLI Control**

In [137]:
#summary statistics PCLI control 
pcli_control_pc02.describe()

Day measured  Buoyant Weight (g)  Tissue Surface Area (mm2)  \
count     31.000000           40.000000                  40.000000   
mean      29.806452           11.628625                 785.054388   
std       24.976548            2.289714                 269.002052   
min        0.000000            7.960000                 308.829232   
25%        0.000000            9.757750                 629.532111   
50%       28.000000           11.128000                 749.871123   
75%       56.000000           13.636750                 985.118770   
max       84.000000           15.567000                1314.561572   

       Surface Area Density (g/mm2)  Calcification rate (mgCaCO3/cm2/day)  \
count                     40.000000                             30.000000   
mean                      67.474308                              6.594643   
std                       18.687758                              4.964867   
min                       24.662464                             -3.350000   
25%                       53.983921                              2.337202   
50%                       70.115341                              8.160714   
75%                       79.049079                             10.143750   
max                      107.004667                             15.714286   

       Tissue growth (mm2/day)  Percent change Surface Area Density  Elevated  \
count                30.000000                            30.000000      40.0   
mean                  6.437653                            29.375779       0.0   
std                   4.432985                            31.239542       0.0   
min                   0.454605                             0.643039       0.0   
25%                   2.925955                             8.015001       0.0   
50%                   4.822425                            12.594612       0.0   
75%                  10.503035                            48.187659       0.0   
max                  16.352839                           108.572254       0.0   

          Sample  
count  40.000000  
mean    4.500000  
std     2.908872  
min     0.000000  
25%     2.000000  
50%     4.500000  
75%     7.000000  
max     9.000000

In [138]:
#missing data pcli control 
print(pcli_control_pc02.isnull().sum()) 

Day measured                             9
Buoyant Weight (g)                       0
Tissue Surface Area (mm2)                0
Surface Area Density (g/mm2)             0
Calcification rate (mgCaCO3/cm2/day)    10
Tissue growth (mm2/day)                 10
Percent change Surface Area Density     10
Elevated                                 0
Sample                                   0
dtype: int64


**PCLI Elevated PCO2**

In [139]:
#summary statistics pcli elevated 
pcli_evelvated_pc02.describe()

Date data taken  Buoyant Weight (g)  Tissue Surface Area (mm2)  \
count        40.000000           40.000000                  40.000000   
mean         42.000000           11.885600                 803.047495   
std          31.703757            1.921558                 257.951876   
min           0.000000            8.613000                 281.061759   
25%          21.000000            9.935000                 649.872267   
50%          42.000000           12.211000                 832.377633   
75%          63.000000           13.359250                 979.464325   
max          84.000000           15.255000                1284.154550   

       Surface Area Density (g/mm2)  Calcification rate (mgCaCO3/cm2/day)  \
count                     40.000000                             20.000000   
mean                      67.880508                              3.790952   
std                       20.185155                              4.463225   
min                       25.952940                             -5.433333   
25%                       54.673972                              1.447024   
50%                       71.010806                              3.535714   
75%                       80.591935                              6.781250   
max                      100.829153                             14.166667   

       Tissue growth (mm2/day)  Percent change Surface Area Density  Elevated  \
count                30.000000                            30.000000      40.0   
mean                  7.084215                            33.386608       1.0   
std                   4.310620                            36.482044       0.0   
min                   1.669264                             1.222209       1.0   
25%                   3.514452                             6.591748       1.0   
50%                   5.760531                            16.044810       1.0   
75%                   9.414083                            54.676468       1.0   
max                  16.876889                           146.636266       1.0   

          Sample  
count  40.000000  
mean   14.500000  
std     2.908872  
min    10.000000  
25%    12.000000  
50%    14.500000  
75%    17.000000  
max    19.000000

In [140]:
#missing data pcli summary statistics 
print(pcli_evelvated_pc02.isnull().sum()) 

Date data taken                          0
Buoyant Weight (g)                       0
Tissue Surface Area (mm2)                0
Surface Area Density (g/mm2)             0
Calcification rate (mgCaCO3/cm2/day)    20
Tissue growth (mm2/day)                 10
Percent change Surface Area Density     10
Elevated                                 0
Sample                                   0
dtype: int64
